crates/net/network/src/metrics.rs

```diff
@@ -119,3 +119,28 @@ impl DisconnectMetrics {
         }
     }
 }
+
+/// Metrics for the EthRequestHandler
+#[derive(Metrics)]
+#[metrics(scope = "network")]
+pub struct EthRequestHandlerMetrics {
+    /// Total number of requests received
+    #[allow(unused)]
+    pub(crate) requests_received: Counter,
+
+    /// Total number of requests served
+    #[allow(unused)]
+    pub(crate) requests_processed: Counter,
+
+    /// Total number of requests that failed to be served
+    #[allow(unused)]
+    pub(crate) requests_failed: Counter,
+
+    /// Total number of requests that timed out
+    #[allow(unused)]
+    pub(crate) requests_timed_out: Counter,
+
+    /// Total number of requests that were cancelled
+    #[allow(unused)]
+    pub(crate) requests_cancelled: Counter,
```

Comment:
for now, I think we only want:

* `received_headers_requests`
* `received_bodies_requests`